html/pfappserver/root/vue.config.js

```diff
@@ -4,6 +4,11 @@ module.exports = {
   publicPath: '/admin',
   outputDir: 'dist',
   indexPath: 'index.html',
+  devServer: {
+    port: 8081,
+    https: true,
+    proxy: 'https://API_IP:1443'
```

Comment:
We could pull `VUE_APP_API_BASEURL` but I'm not sure what's the use case for that const, so I was hesitant

Reply:
I understand what `VUE_APP_API_BASEURL` is, I don't understand when would you want to set it. Since trying to serve the UI on a different host will run into CORS issues, I'm not sure what's the point of having `VUE_APP_API_BASEURL` env variable.

`proxy` field is not for specifying the address of the `webpack-dev-server`, it's for specifying where to proxy the request to. We should either use `VUE_APP_API_BASEURL`(since it already points to the URL of the API) or we could use `https://localhost:1443` as the default. Either way, we probably don't want another environment variable. `webpack-dev-server` is usually configured through `vue.config.js`. I don't think having env variables for this helps anyone.

Reply:
Since `VUE_APP_` env's are baked into the builds, they don't help anyone except development anyway.

I'm not a fan of reusing ENV's meant for one function, being used for another. It helps makes the distinction explicit when tracing code-paths.

I've used `VUE_APP_API_BASEURL` to use local sources with a remote PF instance. If the hostname differs then CORS either needs to be disabled or the `Access-Control-Allow-Origin` needs to be set, or a proxy with SSL-termination needs to be used in-the-middle. But if the same hostname is used and only the port is changed then CORS works as-is. Normally I used ssh LocalForward to bind a remote PF_HOST:1443 to a different local port, and use this ENV to change the port w/ the same hostname.

Reply:
> I'm not a fan of reusing ENV's meant for one function, being used for another.

The meaning of `VUE_APP_API_BASEURL` is "the address of the API". It doesn't change based on whether the UI is packaged or not. 

I thought that maybe the `VUE_APP_API_BASEURL` is there in case the UI is deployed in a different machine from the API.
If it's only for developers maybe this const can be removed altogether? `webpack-dev-server` handles the CORS and modifying the `vue.config.json` is the standard way for configuring the `webpack-dev-server`. The downside of that is that we'll modify the config and will have to remember to not push the changes. 

I think the best thing to do is to change `vue.config.json` to use `VUE_APP_API_BASEURL`. Later, it's worth to consider whether we need the usage of `VUE_APP_API_BASEURL` in `html/pfappserver/root/src/utils/api.js`. My guess is no, but it can be done in a different PR

Reply:
NVM. I thought that we can just use the API IP address, since it doesn't change based on whether the UI is packaged or served, but if the UI is served, `html/pfappserver/root/src/utils/api.js` needs to use the defaults. I'll add a new const, but we should consider whether we need the `VUE_APP_API_BASEURL` at all. `webpack-dev-server` hadles CORS and doesn't interfere with production code.